modules/logging-set-input-rate-limit.adoc

```diff
@@ -0,0 +1,77 @@
+// Module included in the following assemblies:
+//
+// * logging/performance_reliability/logging-flow-control-mechanisms.adoc
+
+:_mod-docs-content-type: PROCEDURE
+[id="logging-set-input-rate-limit_{context}"]
+= Configuring log forwarder input rate limits
+
+You can limit the rate of incoming logs that are collected by configuring the `ClusterLogForwarder` custom resource (CR).
+Input limits can be set on a per-container or per-namespace basis.
+
+.Prerequisites
+
+* You have installed the {clo}.
+* You have administrator permissions.
+
+.Procedure
+
+. Add a `maxRecordsPerSecond` limit value to the `ClusterLogForwarder` CR for a specified input.
++
+The following examples shows how to configure input rate limits for different scenarios.
++
+.Example `ClusterLogForwarder` CR that sets a per-container limit for containers with certain labels
+[source,yaml]
+----
+apiVersion: logging.openshift.io/v1
+kind: ClusterLogForwarder
+metadata:
+# ...
+spec:
+# ...
+  inputs:
+    - name: <input_name> <1>
```

Comment:
For your long term thinking about structure: The log selection (labels, namespace) and rate limiting features are independent. We should be able to explain how to use labels &  namespaces to select a set of logs in one place, and explain how to apply a rate limit to all the logs selected by an input in another. It may make sense to provide complete examples like this from time to time, but we need to be careful of getting into a habit of redundantly explaining every possible combination of features with each feature.

Reply:
I understand what you're saying. I think it can be handled on a case by case basis depending on the user story. Some things as you've said I believe are more useful with complete examples - I think this is one of those times - whereas other features may be fine just shown in their simplest form. I think this is something we can work out as we develop a more full doc set for `inputs` generally, since we're sort of breaking new ground documenting these for right now 🙂 continuous improvement FTW